metric/meter.go

```diff
@@ -142,9 +142,9 @@ type Observer interface {
 	embedded.Observer
 
 	// ObserveFloat64 records the float64 value with attributes for obsrv.
-	ObserveFloat64(obsrv instrument.Float64Observable, value float64, attributes ...attribute.KeyValue)
+	ObserveFloat64(obsrv instrument.ObservableT[float64], value float64, attributes ...attribute.KeyValue)
 	// ObserveInt64 records the int64 value with attributes for obsrv.
-	ObserveInt64(obsrv instrument.Int64Observable, value int64, attributes ...attribute.KeyValue)
+	ObserveInt64(obsrv instrument.ObservableT[int64], value int64, attributes ...attribute.KeyValue)
```

Comment:
Not clear what advantage has this design where you ask users to call this vs asking users to call "Observe" on the `ObservableT` directly, since they have to have the `ObservableT` instances when registering the callback.

If we do that, we don't need `ObservableT` to be public anymore.

Reply:
The types implementing `ObservableT` [do not have and `Observe` method anymore](https://github.com/open-telemetry/opentelemetry-go/pull/3586). This was a follow up to a decision to [refactor the `RegisterCallback` API](https://github.com/open-telemetry/opentelemetry-go/pull/3584) so it can provide the guarantees the specification [recommends](https://github.com/open-telemetry/opentelemetry-go/issues/3380#issuecomment-1374311353) and [requires](https://github.com/open-telemetry/opentelemetry-go/issues/3373#issuecomment-1371605228).

The other way of allowing users to directly call `Observe` did not clearly restrict when they can call it and therefore provided potential for unintentional bugs from the user. It also wasn't a design we could make compliant with the specification given we couldn't completely restrict that method to only be called in a callback registered with the instrument.

Reply:
I missed the `Observe` removing part, was familiar with Java where they have that. Makes sense then the overall design, just make sure you are allowed to extend the Callback interface to allow recording more "types" in the future.